synapse/handlers/register.py

```diff
@@ -217,8 +217,8 @@ def register_user(
 
         else:
             # autogen a sequential user ID
-            user = None
-            while not user:
+            # Fail after being unable to find a suitable ID a few times
+            for x in range(10):
```

Comment:
This won't raise an exception if it runs out, and so registration will continue with a `None` user_id. You probably want to keep the while, but have a counter that if bigger than 10 raises an exception